### PR TITLE
Add support for Python 3.11 and remove support for 3.8

### LINKED ISSUE #646: Support Python 3.11; remove support for 3.8
May be worth waiting for the 3.11.1 release before making this change, to avoid too many teething issues.

### PR DESCRIPTION
**Pull request by cqc-alec**
Closes #646 .

Will modify names of required CI checks if approved.

Release workflow tested [here](https://github.com/CQCL/tket/actions/runs/3759526258).

### CODE REVIEW

File: pytket/tests/requirements.txt [hunk 1]
```diff
@@ -7,6 +7,6 @@ py
 hypothesis
 docker
 jsonschema
-quimb
+quimb;python_version<"3.11"
```
Comment (cqc-melf) [inline pytket/tests/requirements.txt:10]: Do you know if they have any plans for 3.11 support in the near future?

**Review by cqc-melf — COMMENTED** [review]
What is our plan for the extensions? Do we know if there are any bigger issues expected with 3.11?

File: pytket/tests/requirements.txt (hunk 1 above)
Comment (cqc-alec) [reply pytket/tests/requirements.txt:10]: I would expect so, since it seems to be quite well maintained.

**Comment by cqc-alec** [comment]
> What is our plan for the extensions? Do we know if there are any bigger issues expected with 3.11?

We will test the extensions individually with Python 3.11; I expect most will be fine but there may be a few for which we have to delay 3.11 support.

**Review by cqc-melf — COMMENTED** [review]
I think Build and test / Build and test (ubuntu-22.04, 3.10) (pull_request) should be fixed before merging this?

**Comment by cqc-alec** [comment]
> I think Build and test / Build and test (ubuntu-22.04, 3.10) (pull_request) should be fixed before merging this?

Yes, this is a separate issue but I'm working on it now...

**Review by cqc-melf — APPROVED** [review]
Feel free to merge it without the 3.8 checks

**Pull request merged**